db/db_iter.cc

```diff
@@ -674,6 +692,11 @@ void DBIter::PrevInternal() {
       }
       return;
     }
+
+    if (TooManyInternalKeysSkipped()) {
+      return;
+    }
+
```

Comment:
These lines are not necessary since you set `valid_ = false;` already in `TooManyInternalKeysSkipped`. So line 700 will break;